packages/office-ui-fabric-react/src/components/Checkbox/Checkbox.base.tsx

```diff
@@ -28,18 +28,15 @@ export class CheckboxBase extends React.Component<ICheckboxProps, ICheckboxState
   private _id: string;
   private _classNames: { [key in keyof ICheckboxStyles]: string };
 
-  public static getDerivedStateFromProps(props: ICheckboxProps, state: ICheckboxState): ICheckboxState | null {
-    if (!props.defaultIndeterminate && state.isIndeterminate) {
-      return {
-        isIndeterminate: !!props.indeterminate
-      };
+  public static getDerivedStateFromProps(nextProps: ICheckboxProps, prevState: ICheckboxState): ICheckboxState | null {
+    const state: Partial<ICheckboxState> = {};
+    if (nextProps.indeterminate !== undefined) {
+      prevState.isIndeterminate = !!nextProps.indeterminate;
     }
-    if (props.checked !== undefined) {
-      return {
-        isChecked: !!props.checked
-      };
+    if (nextProps.checked !== undefined) {
+      prevState.isChecked = !!nextProps.checked;
     }
-    return null;
+    return Object.keys(state).length ? state : null;
```

Comment:
>state [](start = 23, length = 5)

i might be missing something but where in between line 32 and 39 is state actually modified? wouldn't this always return null?

Reply:
Also, if there is indeed a bug here, should the tests have caught something?

---
In reply to: [335074857](https://github.com/OfficeDev/office-ui-fabric-react/pull/10644#discussion_r335074857) [](ancestors = 335074857)

Reply:
Yes, that's meant to be `state`. Not sure how I missed that...

Reply:
I'm wondering if it appeared to work because the mutated old state object was being preserved...

Reply:
[https://codepen.io/cliffkoh/pen/MWWeamj?editors=1010](https://codepen.io/cliffkoh/pen/MWWeamj?editors=1010) confirmed that you can mutate state and it would actually stick

---
In reply to: [335121554](https://github.com/OfficeDev/office-ui-fabric-react/pull/10644#discussion_r335121554) [](ancestors = 335121554)